src/ripple/app/tx/impl/details/NFTokenUtils.cpp

```diff
@@ -636,5 +674,122 @@ deleteTokenOffer(ApplyView& view, std::shared_ptr<SLE> const& offer)
     return true;
 }
 
+bool
+repairNFTokenDirectoryLinks(ApplyView& view, AccountID const& owner)
+{
+    bool didRepair = false;
+
+    auto const last = keylet::nftpage_max(owner);
+
+    std::shared_ptr<SLE> page = view.peek(Keylet(
+        ltNFTOKEN_PAGE,
+        view.succ(keylet::nftpage_min(owner).key, last.key.next())
+            .value_or(last.key)));
+
+    if (!page)
+        return didRepair;
+
+    if (page->key() == last.key)
+    {
+        // There's only one page in this entire directory.  There should be
+        // no links on that page.
+        bool const nextPresent = page->isFieldPresent(sfNextPageMin);
+        bool const prevPresent = page->isFieldPresent(sfPreviousPageMin);
+        if (nextPresent || prevPresent)
+        {
+            didRepair = true;
+            if (prevPresent)
+                page->makeFieldAbsent(sfPreviousPageMin);
+            if (nextPresent)
+                page->makeFieldAbsent(sfNextPageMin);
+            view.update(page);
+        }
+        return didRepair;
+    }
+
+    // First page is not the same as last page.  The first page should not
+    // contain a previous link.
+    if (page->isFieldPresent(sfPreviousPageMin))
+    {
+        didRepair = true;
+        page->makeFieldAbsent(sfPreviousPageMin);
+        view.update(page);
```

Comment:
I remember Nik mentioned that the nodes at the end of the doubly linked list should point to the other end. Would we want to make `sfPreviousPageMin` of the first page point to the `last` page? 

Nik's comment:
https://github.com/XRPLF/rippled/pull/4346#discussion_r1026600781

Though I don't think this is actually enforced in the current NFTPage code...

Reply:
Hi @shawnxie999, great question.  There have definitely been times that I've wished that the first and last links in the list linked to each other.  But they don't and, as far as I know, never have.

The spec (https://github.com/XRPLF/XRPL-Standards/discussions/46) does not give any specifics about how `sfPreviousPageMin` and `NextPageMin` are handled.  So I think this is a case where "the code is the spec".  The existing code does not link the back of the list to the front of the list.  And the existing unit tests do not expect such links to exist.  This amendment does not, and should not, change that behavior.